launch/gz_components.launch.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2024 Husarion sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import yaml
+from ament_index_python.packages import get_package_share_directory
+from launch import LaunchDescription
+from launch.actions import (
+    DeclareLaunchArgument,
+    IncludeLaunchDescription,
+    OpaqueFunction,
+)
+
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.substitutions import LaunchConfiguration, EnvironmentVariable
+
+
+def get_value(node: yaml.Node, key: str):
+    try:
+        value = node[key]
+        if value == "None":
+            value = ""
+        return value
+
+    except KeyError:
+        return ""
+
+
+def get_launch_description(name: str, package: str, namespace: str, component: yaml.Node):
+    return IncludeLaunchDescription(
+        PythonLaunchDescriptionSource([package, "/launch/gz_", name, ".launch.py"]),
+        launch_arguments={
+            "robot_namespace": namespace,
+            "device_namespace": get_value(component, "namespace"),
+            "tf_prefix": get_value(component, "tf_prefix"),
+            "gz_bridge_name": component["namespace"][1:] + "_gz_bridge",
+            "camera_name": get_value(component, "name"),
+        }.items(),
+    )
+
+
+def get_launch_descriptions_from_yaml_node(
+    node: yaml.Node, package: os.PathLike, namespace: str
+) -> IncludeLaunchDescription:
+    actions = []
+    for component in node["components"]:
+        if component["type"] == "LDR01" or component["type"] == "LDR06":
+            actions.append(get_launch_description("slamtec_rplidar", package, namespace, component))
+
+        if component["type"] == "LDR13":
+            actions.append(get_launch_description("ouster_os", package, namespace, component))
+
+        if component["type"] == "LDR20":
+            actions.append(get_launch_description("velodyne", package, namespace, component))
+
+        if component["type"] == "CAM01":
+            actions.append(get_launch_description("orbbec_astra", package, namespace, component))
+
```

Comment:
Make it easier to maintain
```suggestion
    component_to_launch = {
        "LDR01": "slamtec_rplidar",
        "LDR06": "slamtec_rplidar",
        "LDR13": "ouster_os",
        "LDR20": "velodyne",
        "CAM01": "orbbec_astra"
    }

    for component in node["components"]:
        component_type = component["type"]
        if component_type in component_to_launch:
            launch_description = get_launch_description(component_to_launch[component_type], package, namespace, component)
            actions.append(launch_description)
```

Reply:
Nah, I prefer `get_launch_description` to stay.